lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/unmanaged/UnmanagedAmYarnSubmissionHelper.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.yarn.client.unmanaged;
+
+import org.apache.hadoop.security.token.Token;
+import org.apache.hadoop.yarn.api.records.*;
+import org.apache.hadoop.yarn.client.api.YarnClient;
+import org.apache.hadoop.yarn.client.api.YarnClientApplication;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import org.apache.hadoop.yarn.exceptions.YarnException;
+import org.apache.hadoop.yarn.security.AMRMTokenIdentifier;
+import org.apache.reef.runtime.yarn.client.SecurityTokenProvider;
+import org.apache.reef.runtime.yarn.client.UserCredentialSecurityTokenProvider;
+import org.apache.reef.runtime.yarn.util.YarnTypes;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Helper code that wraps the YARN Client API for our purposes.
+ */
+final class UnmanagedAmYarnSubmissionHelper implements AutoCloseable {
+
+  private static final Logger LOG = Logger.getLogger(UnmanagedAmYarnSubmissionHelper.class.getName());
+
+  private final SecurityTokenProvider tokenProvider;
+  private final YarnClient yarnClient;
+  private final ApplicationSubmissionContext applicationSubmissionContext;
+  private final ApplicationId applicationId;
+
+  UnmanagedAmYarnSubmissionHelper(final YarnConfiguration yarnConfiguration,
```

Comment:
@jwang98052 You can use either one, but using the specialized Unmanaged AM configuration results in much simpler client code. You'll see that in the subsequent PRs with the Unmanaged AM HelloREEF app. Still, I want to keep both variants - this way the user can quickly switch the app between two modes and still have access to a short user-friendly API